### ISSUE TITLE
`TypeAliasType` with `Concatenate` + `ParamSpec` resolves to `Unknown`, while native type statement works correctly

### ISSUE THREAD

**Issue by jiang-zhexin** [astral-sh/ty#3683]
### Description

I discovered this while using **altair**: `alt.Chart().mark_point()` was inferred as `Unknown` by `ty`. After investigation, I traced the root cause to a `decorate` function in altair's codebase:
https://github.com/vega/altair/blob/main/altair/utils/core.py#L768

This `decorate` function uses `TypeAliasType` with `Concatenate` + `ParamSpec` to define wrapper type aliases, which `ty` fails to resolve, in turn causing wrapped methods like `mark_point()` to become `Unknown`.

Below is a minimal reproduction extracted from that pattern.

### Minimal Reproduction

`type-issue.py` (uses `TypeAliasType` — broken):

```python
from collections.abc import Callable
from typing import TYPE_CHECKING, Concatenate, ParamSpec, TypeAliasType, TypeVar, reveal_type

T = TypeVar("T")
P = ParamSpec("P")
R = TypeVar("R")

WrapsMethod = TypeAliasType(
    "WrapsMethod", Callable[Concatenate[T, ...], R], type_params=(T, R)
)
WrappedMethod = TypeAliasType(
    "WrappedMethod", Callable[Concatenate[T, P], R], type_params=(T, P, R)
)

def decorate(cb: WrapsMethod[T, R], /) -> WrappedMethod[T, P, R]:
    # _wrap_and_copy_doc(tp, cb)
    return cb

@decorate
def myfunc(input: int) -> int:
    return 1

if TYPE_CHECKING:
    reveal_type(myfunc)
```

`type-issue-3-12.py` (uses `native type` — works):

```python
from collections.abc import Callable
from typing import TYPE_CHECKING, Concatenate, reveal_type

type WrapsMethod[T, R] = Callable[Concatenate[T, ...], R]
type WrappedMethod[T, **P, R] = Callable[Concatenate[T, P], R]

def decorate[T, **P, R](cb: WrapsMethod[T, R], /) -> WrappedMethod[T, P, R]:
    # _wrap_and_copy_doc(tp, cb)
    return cb

@decorate
def myfunc(input: int) -> int:
    return 1

if TYPE_CHECKING:
    reveal_type(myfunc)
```

### Results

| Checker | TypeAliasType                               | native type                                 |
| ------- | ------------------------------------------- | ------------------------------------------- |
| pyright | `(int, **P@decorate) -> int`                | `(int, **P@decorate) -> int`                |
| mypy    | `def [P] (int, *P.args, **P.kwargs) -> int` | `def [P] (int, *P.args, **P.kwargs) -> int` |
| ty      | `Unknown  `                                 | `[**P'return](int, /, ...) -> int`          |

### Real-world impact

This affects `altair` users who rely on `ty` for type checking — methods decorated via `altair/utils/core.py#decorate` (e.g., `mark_point`, `mark_line`, etc.) are all inferred as `Unknown`.

### Version

0.0.44

**Comment by dhruvmanila** [comment]
Thanks for the report.

ty doesn't support generic manual type aliases yet like the one constructed here using `TypeAliasType`, refer to https://github.com/astral-sh/ty/issues/1737. So, I'll consider this a duplicate of the original issue.
